src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -29,10 +29,12 @@ internal partial class SyntaxTreeSemanticModel : CSharpSemanticModel
 
         private static readonly Func<CSharpSyntaxNode, bool> isMemberDeclarationFunction = IsMemberDeclaration;
 
-        internal SyntaxTreeSemanticModel(CSharpCompilation compilation, SyntaxTree syntaxTree)
+        internal SyntaxTreeSemanticModel(CSharpCompilation compilation, SyntaxTree syntaxTree, bool suppressAccessChecks = false)
         {
             this.compilation = compilation;
             this.syntaxTree = syntaxTree;
+            this.HasAccessChecksSuppressed = suppressAccessChecks;
```

Comment:
Please use the pattern as for other things; make this a local field in only those subclasses that need it, and make the original property a virtual (you can make the base implemenetation return false).